xocto/ranges.py

```diff
@@ -837,6 +837,14 @@ def __init__(self, start: datetime.datetime, end: datetime.datetime):
         """
         super().__init__(start, end, boundaries=RangeBoundaries.INCLUSIVE_EXCLUSIVE)
 
+    def __lt__(self, other: Range[datetime.datetime]) -> bool:
+        if isinstance(other, FiniteDatetimeRange):
```

Comment:
🤔 I think this could be optimised further to:

```py3
if other.start is None:
    # We don't need to check anything more if the other range
    # is open-ended
    return False
else:
    return self.start < other.start
```

That should be faster as I think that checking for `None` is quicker than an `isinstance` check.